tailcfg/tailcfg.go

```diff
@@ -2455,6 +2455,25 @@ type HealthChangeRequest struct {
 	NodeKey key.NodePublic
 }
 
+// SetDeviceAttributesRequest is a request to update the
+// current node's attributes.
+//
+// As of 2024-12-30, this is an experimental dev feature
+// for internal testing. See tailscale/corp#24690.
+type SetDeviceAttributesRequest struct {
+	Version CapabilityVersion
+	NodeKey key.NodePublic
+
+	// Update is a map of attributes to update.
```

Comment:
I suspect at some point we might want clients to be able to set an expiry time, like our HTTP API supports: https://tailscale.com/api#tag/devices/POST/device/{deviceId}/attributes/{attributeKey}

Would that be another dict of key->time.Time? (similar to the `expiries` field returned by our [get api](https://tailscale.com/api#tag/devices/GET/device/{deviceId}/attributes))

Or should the API be mapping keys to another struct, which would contain value alongside an optional expiry time?

Reply:
How about we keep it any-to-any now (where the value `any` can be float64/bool/string/nil) and in the future we could add a struct (JSON object) type for other metadata like expiries?